test/spec/modules/ucfunnelBidAdapter_spec.js

```diff
@@ -0,0 +1,114 @@
+import { expect } from 'chai';
+import { spec } from 'modules/ucfunnelBidAdapter';
+
+const URL = '//hb.aralego.com/header';
+const BIDDER_CODE = 'ucfunnel';
+const validBidReq = {
+  bidder: BIDDER_CODE,
+  params: {
+    adid: 'test-ad-83444226E44368D1E32E49EEBE6D29'
+  },
+  sizes: [[300, 250]],
+  bidId: '263be71e91dd9d',
+  auctionId: '9ad1fa8d-2297-4660-a018-b39945054746',
+};
+
+const invalidBidReq = {
+  bidder: BIDDER_CODE,
+  params: {
+    adid: 123456789
+  },
+  sizes: [[300, 250]],
+  bidId: '263be71e91dd9d',
+  auctionId: '9ad1fa8d-2297-4660-a018-b39945054746'
+};
+
+const bidReq = [{
+  bidder: BIDDER_CODE,
+  params: {
+    adid: 'test-ad-83444226E44368D1E32E49EEBE6D29'
+  },
+  sizes: [[300, 250]],
+  bidId: '263be71e91dd9d',
+  auctionId: '9ad1fa8d-2297-4660-a018-b39945054746'
+}];
+
+const validBidRes = {
+  ad_id: 'ad-83444226E44368D1E32E49EEBE6D29',
+  adm: '<html style="height:100%"><body style="width:300px;height: 100%;padding:0;margin:0 auto;"><div style="width:100%;height:100%;display:table;"><div style="width:100%;height:100%;display:table-cell;text-align:center;vertical-align:middle;"><a href="//www.ucfunnel.com/" target="_blank"><img src="//cdn.aralego.net/ucfad/house/ucf/AdGent-300x250.jpg" width="300px" height="250px" align="middle" style="border:none"></a></div></div></body></html>',
+  cpm: 0.01,
+  height: 250,
+  width: 300
+};
+
+const bidResponse = validBidRes;
+
+const bidResArray = [
+  validBidRes,
+  {
+    ad: '',
+    bidRequestId: '263be71e91dd9d',
+    cpm: 100,
+    adId: '123abc',
+    currency: 'USD',
+    netRevenue: true,
+    width: 300,
+    height: 250,
+    ttl: 360
+  },
+  {
+    ad: '<div>Hello</div>',
+    bidRequestId: '',
+    cpm: 0,
+    adId: '123abc',
+    currency: 'USD',
+    netRevenue: true,
+    width: 300,
+    height: 250,
+    ttl: 360
+  }
+];
+
+describe('ucfunnel Adapter', () => {
+  describe('request', () => {
+    it('should validate bid request', () => {
+      expect(spec.isBidRequestValid(validBidReq)).to.equal(true);
+    });
+    it('should not validate incorrect bid request', () => {
+      expect(spec.isBidRequestValid(invalidBidReq)).to.equal(false);
+    });
+  });
+  describe('build request', () => {
+    it('bidRequest method', () => {
+      const request = spec.buildRequests(bidReq);
+      expect(request[0].method).to.equal('GET');
+    });
+    it('bidRequest url', () => {
+      const request = spec.buildRequests(bidReq);
+      expect(request[0].url).to.equal(URL);
+    });
+    it('bidRequest data', () => {
+      const request = spec.buildRequests(bidReq);
```

Comment:
There's no reason this needs to be three separate tests.  It will just make the test suite run longer for no discernible reason.